traitsui/tests/editors/test_liststr_editor_selection.py

```diff
@@ -74,21 +92,539 @@ class ListStrEditorWithSelectedIndex(HasTraits):
 )
 
 
-def get_selected(control):
+def get_selected_indices(editor):
     """ Returns a list of the indices of all currently selected list items.
     """
-    import wx
+    if is_current_backend_wx():
+        import wx
+
+        item = -1
+        selected = []
+        while True:
+            item = editor.control.GetNextItem(
+                item, wx.LIST_NEXT_ALL, wx.LIST_STATE_SELECTED
+            )
+            if item == -1:
+                break
+            selected.append(item)
+        return selected
+
+    elif is_current_backend_qt4():
+        indices = editor.list_view.selectionModel().selectedRows()
+        return [i.row() for i in indices]
+
+    else:
+        raise unittest.SkipTest("Test not implemented for this toolkit")
+
+
+def set_selected_single(editor, index):
+    if is_current_backend_wx():
+        clear_selection(editor)
+        editor.control.Select(index)
+
+    elif is_current_backend_qt4():
+        from pyface.qt.QtGui import QItemSelectionModel
+
+        smodel = editor.list_view.selectionModel()
+        mi = editor.model.index(index)
+        smodel.select(mi, QItemSelectionModel.ClearAndSelect)
+
+    else:
+        raise unittest.SkipTest("Test not implemented for this toolkit")
+
+
+def set_selected_multiple(editor, indices):
+    if is_current_backend_wx():
+        clear_selection(editor)
+        for index in indices:
+            editor.control.Select(index)
+
+    elif is_current_backend_qt4():
+        from pyface.qt.QtGui import QItemSelectionModel
+
+        smodel = editor.list_view.selectionModel()
+        smodel.clearSelection()
```

Comment:
No, in case of single selection there is no need to manually clear the selection, it is integrated into toolkit functions (will modify wx as mentioned above).
In case of `multi_select` clearing the selection and selecting new indices is the same user action (a new selection "session" while holding Ctrl) but clearing has to be done manually (see `_multi_selected_indices_changed` of the editors).

Reply:
Hmmm I am still confused, I am reading these two lines:
```
smodel = editor.list_view.selectionModel()
smodel.clearSelection()
```
and they look to have the same effect as the line in `clear_selection` apart from the assignment of `smodel`:
```
editor.list_view.selectionModel().clearSelection()
```
Anyway, this is minor.

Reply:
Oh, sorry, I misread the question. My comment answers question below (`set_selected_single` and `set_selected_multiple` being the same). This line IS the same as `clear_selection(editor)`.

Reply:
Cool. Thank you!